src/net.cpp

```diff
@@ -776,7 +791,160 @@ CNetMessage V1TransportDeserializer::GetMessage(const std::chrono::microseconds
     return msg;
 }
 
-void V1TransportSerializer::prepareForTransport(CSerializedNetMsg& msg, std::vector<unsigned char>& header) {
+int V2TransportDeserializer::readHeader(Span<const uint8_t> msg_bytes)
+{
+    // copy data to temporary parsing buffer
+    const unsigned int remaining = CHACHA20_POLY1305_AEAD_AAD_LEN - m_hdr_pos;
+    const unsigned int copy_bytes = std::min<unsigned int>(remaining, msg_bytes.size());
```

Comment:
Any reason to use `unsigned int` for offsets/sizes here and in `readData` instead of `size_t`?

Reply:
That's a good idea. Done.